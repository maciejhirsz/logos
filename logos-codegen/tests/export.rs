#![cfg(feature = "debug")]

use std::path::Path;

use insta::assert_snapshot;

#[test]
fn test_complex() {
    test_export("complex");
}

fn test_export(case: &str) {
    let mut input_path = Path::new(env!("CARGO_MANIFEST_DIR")).join("tests/data/export");
    input_path.push(case);
    input_path.push("input.rs");

    let input = std::fs::read_to_string(input_path).expect("Unable to read input.rs");

    let _ = logos_codegen::generate(input.parse().expect("Unable to parse input.rs"));

    let generated_dot = std::fs::read_to_string(format!("{case}_export_tmp/{case}.dot")).expect("Unable to read dot file");
    let generated_mermaid = std::fs::read_to_string(format!("{case}_export_tmp/{case}.mmd")).expect("Unable to read mermaid file");

    assert_snapshot!(generated_dot);
    assert_snapshot!(generated_mermaid);

    // cleanup
    let _ = std::fs::remove_dir_all(format!("{case}_export_tmp"));
}
