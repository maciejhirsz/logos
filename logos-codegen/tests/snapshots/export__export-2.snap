---
source: logos-codegen/tests/export.rs
expression: generated_mermaid
---
flowchart TB
n1024ctx0["State 1024 (ctx 0)"]
style n1024ctx0 stroke:#00C853
n1536ctx0["State 1536 (ctx 0)"]
style n1536ctx0 stroke:#00C853
n1536ctx0-->|"r"|n7680ctx0
n2048ctx2["State 2048 (ctx 2)"]
style n2048ctx2 stroke:#00C853
n2560ctx2["State 2560 (ctx 2)"]
style n2560ctx2 stroke:#00C853
n2560ctx2-->|"\\x00..=`|{..=\\xff"|n3584ctx3
n2560ctx2-->|"a..=z"|n4096ctx3
n2560ctx2-->|"EOI"|n3584ctx3
n3072ctx2["State 3072 (ctx 2)"]
style n3072ctx2 stroke:#00C853
n3072ctx2-->|"\\x00..=`|{..=\\xff"|n3584ctx3
n3072ctx2-->|"a..=b|d..=z"|n4096ctx3
n3072ctx2-->|"c"|n4608ctx3
n3072ctx2-->|"EOI"|n3584ctx3
n3584ctx3["State 3584 (ctx 3)"]
style n3584ctx3 stroke:#00C853
n4096ctx3["State 4096 (ctx 3)"]
style n4096ctx3 stroke:#00C853
n4096ctx3-->|"\\x00..=`|{..=\\xff"|n3584ctx3
n4096ctx3-->|"a..=z"|n4096ctx3
n4096ctx3-->|"EOI"|n3584ctx3
n4608ctx3["State 4608 (ctx 3)"]
style n4608ctx3 stroke:#00C853
n4608ctx3-->|"\\x00..=`|{..=\\xff"|n3584ctx3
n4608ctx3-->|"a..=s|u..=z"|n4096ctx3
n4608ctx3-->|"t"|n5120ctx3
n4608ctx3-->|"EOI"|n3584ctx3
n5120ctx3["State 5120 (ctx 3)"]
style n5120ctx3 stroke:#00C853
n5120ctx3-->|"\\x00..=`|{..=\\xff"|n5632ctx1
n5120ctx3-->|"a..=z"|n6144ctx1
n5120ctx3-->|"EOI"|n5632ctx1
n5632ctx1["State 5632 (ctx 1)"]
style n5632ctx1 stroke:#00C853
n6144ctx1["State 6144 (ctx 1)"]
style n6144ctx1 stroke:#00C853
n6144ctx1-->|"\\x00..=`|{..=\\xff"|n3584ctx3
n6144ctx1-->|"a..=z"|n4096ctx3
n6144ctx1-->|"EOI"|n3584ctx3
n6656["State 6656"]
style n6656 stroke:#000000
n6656-->|"a..=r|t..=z"|n7168
n6656-->|"s"|n8192
n7168["State 7168"]
style n7168 stroke:#000000
n7168-->|"\\x00..=\\xff"|n1024ctx0
n7168-->|"EOI"|n1024ctx0
n7680ctx0["State 7680 (ctx 0)"]
style n7680ctx0 stroke:#000000
n7680ctx0-->|"\\x00..=`|{..=\\xff"|n2048ctx2
n7680ctx0-->|"a..=t|v..=z"|n2560ctx2
n7680ctx0-->|"u"|n3072ctx2
n7680ctx0-->|"EOI"|n2048ctx2
n8192["State 8192"]
style n8192 stroke:#000000
n8192-->|"\\x00..=s|u..=\\xff"|n1024ctx0
n8192-->|"t"|n1536ctx0
n8192-->|"EOI"|n1024ctx0
